static/src/javascripts/projects/common/modules/experiments/tests/recommended-for-you.js

```diff
@@ -0,0 +1,157 @@
+    'common/utils/template',
+    'common/views/svg',
+    'common/utils/mediator',
+    'common/modules/onward/history',
+    'common/utils/ajax',
+    'text!common/views/experiments/recommended-for-you.html',
+    'inlineSvg!svgs/icon/profile-36',
+    'inlineSvg!svgs/icon/arrow-right',
+    'inlineSvg!svgs/icon/marque-36',
+    'common/utils/fetch'
+], function (
+    bean,
+    fastdom,
+    qwery,
+    $,
+    storage,
+    config,
+    template,
+    svg,
+    mediator,
+    history,
+    ajax,
+    recommendedForYouTemplate,
+    profileIcon,
+    rightArrowIcon,
+    guardianLogo,
+    fetch
+) {
+    return function () {
+        this.id = 'RecommendedForYouRecommendations';
+        this.start = '2016-08-02';
+        this.expiry = '2016-12-23';
+        this.author = 'Joseph Smith';
+        this.description = 'Add a personalised container to fronts';
+        this.audience = 1;
+        this.audienceOffset = 0;
+        this.successMeasure = 'Visit frequency';
+        this.audienceCriteria = 'All users';
+        this.dataLinkNames = '';
+        this.idealOutcome = 'People will visit more often';
+
+        var endpoint = 'http://engine.mobile-aws.guardianapis.com/recommendations';
+        var cachedRecommendationsKey = 'gu.cachedRecommendations';
+        var numberOfRecommendations = 4;
+
+        var $opinionSection;
+        var $recommendedForYouSection;
+
+        this.canRun = function () {
+            $opinionSection = $('#opinion');
+            return config.page.contentType === 'Network Front' && $opinionSection.length;
+        };
+
+        this.variants = [
+            {
+                id: 'user-history',
+                test: function () {
+                    populateRecommendationsContainer();
+                }
+            },
+            {
+                id: 'control',
+                test: function () {}
+            }
+        ];
+
+        function populateRecommendationsContainer() {
+            var recommendations = storage.local.get(cachedRecommendationsKey);
+            if (recommendations && new Date(recommendations.expiry) > new Date()) {
+                insertSection(recommendations.items);
+            } else {
+                var promisedRecommendations = getRemoteRecommendations();
+                promisedRecommendations.then(cacheRecommendations);
+                promisedRecommendations.then(insertSection);
+            }
+        }
+
+        function getRemoteRecommendations() {
+            var reqBody = {
+                'pageSize': numberOfRecommendations,
+                'articles': history.test.getHistory().map(item => item[0])
+            };
+
+            var request = fetch(endpoint, {
+                type: 'json',
+                method: 'post',
+                crossOrigin: true,
+                body: JSON.stringify(reqBody),
+                headers: { 'Content-Type': 'application/json' }
+            });
+
+            return request.then(response =>
```

Comment:
... as discussed, I may have jumped the gun there. If we can't use ES6 notation then I guess switch this back to ES5 rather than everything else to ES6! @guardian/dotcom-platform may have some useful info to add